### PR TITLE
event-generating flapper plugin for end-to-end testing

### PR DESCRIPTION
**Pull request by jyotty**
Since upstream monitoring only refreshes info about a check on state change, we'll use this to constantly generate events as a sort of super-keepalive check.

### CODE REVIEW

**Comment by omgjlk** [comment]
This is different than the host keep alive? Is it providing similar functionality?

**Comment by jyotty** [comment]
@j2sol yes. Keepalive is for sensu client to server; this is for sensu server to the upstream event processor. What we're specifically looking for is if a sensu server stops sending events altogether, or if the upstream aggregator stops processing them, which is a gap in our current monitoring.

**Review by omgjlk — CHANGES_REQUESTED** [review]
I think this is fine to start with, but it needs to support criticality like the rest of the plugins we've been introducing, unless there is good reason not to.

**Comment by jyotty** [comment]
Well, there's no threshold for criticality, it's a binary flip between states. I can add an option to make it flip to warning instead of critical though.

**Comment by omgjlk** [comment]
The criticality option is used to turn what would be a critical thing into just a warning. So it doesn't matter if there is a threshold for criticality in the check itself, the criticality option basically neuters the check to prevent pages from unwanted environments.

**Comment by masteinhauser** [comment]
Criticality still makes sense in the context of this check and the alerting pipeline since notifications differ in critical and warning processing. SiteController will most likely use this check for flipping between `critical` and `ok` as it is currently written, but I could definitely use usage of flipping between `warning` and `ok` also.

tl;dr `-z` / `--criticality` should be supported.

**Review by omgjlk — APPROVED** [review]
All good here now.
